### PR TITLE
feat(global): adding task to subset name

### PR DESCRIPTION
**Pull request by jezscha**
adding it in-between family and subset

### CODE REVIEW

File: avalon/tools/creator/app.py [hunk 1]
```diff
@@ -300,6 +300,8 @@ def _on_data_changed(self):
         if asset:
             # Get plugin and family
             plugin = item.data(PluginRole)
+            task = io.Session.get('AVALON_TASK', None)
```
Comment (antirotor) [inline avalon/tools/creator/app.py]: Task cannot be None because next line will then fail.

File: avalon/tools/creator/app.py [hunk 2]
```diff
@@ -301,15 +302,29 @@ def _on_data_changed(self):
             # Get plugin and family
             plugin = item.data(PluginRole)
             family = plugin.family.rsplit(".", 1)[-1]
+            regex = "{}*".format(family)
+            existed_subset_split = family
+
+            if family in self.taskSubsetFamilies:
+                task = io.Session.get('AVALON_TASK', '')
+                sanitized_task = re.sub('[^0-9a-zA-Z]+', '', task)
```
Comment (mkolar) [inline avalon/tools/creator/app.py]: we can replace this with `\d`

Comment (jezscha) [reply avalon/tools/creator/app.py]: No it cannot. This way with \d it is not getting anything else but numbers.. Didn't find shorter equivalent to this.. Closest was `\D` but it is not getting `_` so I will keep this as it is..

Comment (iLLiCiTiT) [reply avalon/tools/creator/app.py]: I think `[^\W_]+` might work but this is maybe more readable

**Comment by jezscha** [comment]
I guess this is ready to merge ;)

**Pull request merged**